libsoftwarecontainer/component-test/filegateway_componenttest.cpp

```diff
@@ -37,27 +38,54 @@ class FileGatewayTest : public SoftwareContainerGatewayTest
 
         gw = std::unique_ptr<FileGateway>(new FileGateway(m_container));
 
-        // Create file
-        ASSERT_TRUE(writeToFile(FILE_PATH, FILE_CONTENT));
+        // Create files
+        mkdir(DIR_PATH.c_str(), 775);
+        assert(createFile(FILE_PATH, FILE_CONTENT));
+        assert(createFile(LONG_FILE_PATH, FILE_CONTENT_2));
+        assert(createFile(FILE_PATH_W_SPACES, FILE_CONTENT_2));
     }
 
     void TearDown() override
     {
         // Remove file
         unlink(FILE_PATH.c_str());
+        unlink(LONG_FILE_PATH.c_str());
+        unlink(FILE_PATH_W_SPACES.c_str());
+        rmdir(DIR_PATH.c_str());
 
         SoftwareContainerGatewayTest::TearDown();
     }
 
+    bool createFile(const std::string &path, const std::string &content)
+    {
+        bool ret = true;
+        std::ofstream out(path);
+        if (out.is_open()) {
+            out << content;
+            if (!out.good()) {
+                ret = false;
+            }
+            out.close();
+        } else {
+            ret = false;
+        }
+        return ret;
+    }
+
     const std::string FILE_CONTENT = "ahdkhqweuyreqiwenomndlaskmd";
     const std::string FILE_PATH = "/tmp/filename.txt";
+    const std::string DIR_PATH = "/tmp/testdir/";
+    const std::string LONG_FILE_PATH = DIR_PATH + "filename.txt";
+    const std::string FILE_CONTENT_2 = "bgsajbjkbjherkgbjhsbnjdfnjk";
+    const std::string FILE_W_SPACES = "example file name.txt";
+    const std::string FILE_PATH_W_SPACES = "/tmp/example file name.txt";
```

Comment:
this looks like filename with spaces to me.  Adding space after tmp would be more explanatory.
`/tmp /example file/name.txt`